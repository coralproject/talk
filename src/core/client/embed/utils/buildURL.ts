export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && search[0] !== "?") {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
