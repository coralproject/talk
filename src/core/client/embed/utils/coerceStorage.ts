import { createInMemoryStorage } from "coral-framework/lib/storage";

function coerceStorage(type: "localStorage" | "sessionStorage"): Storage {
  let storage: Storage;
  try {
    storage = window[type];
  } catch {
    // eslint-disable-next-line no-console
    console.debug(`${type} not available, use in-memory replacement`);

    storage = createInMemoryStorage();
  }

  return storage;
}

export default coerceStorage;
