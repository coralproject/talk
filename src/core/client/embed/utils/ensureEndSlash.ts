export default function ensureEndSlash(p: string) {
  return p.match(/\/$/) ? p : `${p}/`;
}
