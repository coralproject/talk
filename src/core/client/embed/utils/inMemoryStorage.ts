class InMemoryStorage implements Storage {
  private readonly data = new Map<string, string>();

  public get length() {
    return this.data.size;
  }

  public clear() {
    this.data.clear();
  }

  public key(n: number) {
    const keys = [...this.data.keys()];
    if (keys.length < n || n < 0) {
      return null;
    }

    return keys[n];
  }

  public getItem(key: string): string | null {
    return this.data.get(key) || null;
  }

  public removeItem(key: string): void {
    this.data.delete(key);
  }

  public setItem(key: string, value: string): void {
    this.data.set(key, value);
  }
}

export default InMemoryStorage;
