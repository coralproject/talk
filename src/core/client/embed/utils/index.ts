export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
