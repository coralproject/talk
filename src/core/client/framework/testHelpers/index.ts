export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export { default as createSinonStub } from "./createSinonStub";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
export { default as createUUIDGenerator } from "./createUUIDGenerator";
export * from "./denormalize";
export { default as replaceHistoryLocation } from "./replaceHistoryLocation";
