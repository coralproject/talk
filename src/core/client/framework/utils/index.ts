export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
