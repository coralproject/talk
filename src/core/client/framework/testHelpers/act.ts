import TestRenderer from "react-test-renderer";

import { isPromiseLike } from "coral-common/utils";

export default function act<T>(callback: () => T): T {
  let callbackResult: T;
  const actResult = TestRenderer.act(() => {
    callbackResult = callback();
    return callbackResult as any;
  });
  if (isPromiseLike(callbackResult!)) {
    // Return it this way, to preserve warnings that React emits.
    return {
      then(resolve: (value: any) => void, reject: (err?: Error) => void) {
        (actResult as PromiseLike<any>).then(() => {
          resolve(callbackResult);
        }, reject);
      },
    } as any;
  }
  return callbackResult!;
}
