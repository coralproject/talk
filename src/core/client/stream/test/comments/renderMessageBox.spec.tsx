import sinon from "sinon";

import { pureMerge } from "talk-common/utils";
import { waitForElement, within } from "talk-framework/testHelpers";

import { commenters, settings, storyWithNoComments } from "../fixtures";
import create from "./create";

async function createTestRenderer(
  data: {
    story?: any;
    settings?: any;
    loggedIn?: boolean;
  } = {},
  options: { muteNetworkErrors?: boolean } = {}
) {
  const resolvers = {
    Query: {
      settings: sinon.stub().returns(pureMerge(settings, data.settings)),
      viewer: sinon.stub().returns((data.loggedIn && commenters[0]) || null),
      story: sinon.stub().callsFake((_: any, variables: any) => {
        expectAndFail(variables.id).toBe(storyWithNoComments.id);
        return pureMerge(storyWithNoComments, data.story);
      }),
    },
  };

  const { testRenderer, context } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    muteNetworkErrors: options.muteNetworkErrors,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue(storyWithNoComments.id, "storyID");
      localRecord.setValue(Boolean(data.loggedIn), "loggedIn");
    },
  });

  await waitForElement(() =>
    within(testRenderer.root).getByTestID("comments-stream-log")
  );

  const tabPane = await waitForElement(() =>
    within(testRenderer.root).getByTestID("current-tab-pane")
  );

  return {
    testRenderer,
    context,
    tabPane,
  };
}

it("renders message box when not logged in", async () => {
  const { tabPane } = await createTestRenderer({
    story: {
      settings: {
        messageBox: {
          enabled: true,
          icon: "chat",
          content: "**What do you think**?",
        },
      },
    },
  });
  expect(within(tabPane).toJSON()).toMatchSnapshot();
});

it("renders message box when logged in", async () => {
  const { tabPane } = await createTestRenderer({
    story: {
      settings: {
        messageBox: {
          enabled: true,
          icon: "chat",
          content: "**What do you think**?",
        },
      },
    },
    loggedIn: true,
  });
  expect(within(tabPane).toJSON()).toMatchSnapshot();
});

it("renders message box when commenting disabled", async () => {
  const { tabPane } = await createTestRenderer({
    story: {
      settings: {
        messageBox: {
          enabled: true,
          icon: "chat",
          content: "**What do you think**?",
        },
      },
    },
    settings: {
      disableCommenting: {
        enabled: true,
        message: "Commenting disabled",
      },
    },
  });
  expect(within(tabPane).toJSON()).toMatchSnapshot();
});

it("renders message box when story isClosed", async () => {
  const { tabPane } = await createTestRenderer({
    story: {
      isClosed: true,
      settings: {
        messageBox: {
          enabled: true,
          icon: null,
          content: "**What do you think**?",
        },
      },
    },
  });
  expect(within(tabPane).toJSON()).toMatchSnapshot();
});
