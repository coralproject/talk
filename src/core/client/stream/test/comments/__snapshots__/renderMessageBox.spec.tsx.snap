// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders message box when commenting disabled 1`] = `
<section
  aria-labelledby="tab-COMMENTS"
  className="App-tabContent"
  data-testid="current-tab-pane"
  id="tabPane-COMMENTS"
  role="tabpanel"
>
  <div
    className="HorizontalGutter-root Stream-root HorizontalGutter-double"
  >
    <div
      className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter"
    >
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Sign in
      </button>
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantOutlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
    <div
      className="HorizontalGutter-root HorizontalGutter-double"
    >
      <div
        className="CallOut-root CallOut-colorRegular CallOut-fullWidth PostCommentFormClosedSitewide-root"
      >
        Commenting disabled
      </div>
      <div
        className="MessageBox-root"
      >
        <span
          aria-hidden="true"
          className="Icon-root Icon-md MessageBoxIcon-root"
        >
          chat
        </span>
        <div
          className="Markdown-root MessageBoxContent-root"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong>What do you think</strong>?</p>
",
            }
          }
        />
      </div>
    </div>
    <div
      aria-live="polite"
      className="HorizontalGutter-root HorizontalGutter-full"
      data-testid="comments-stream-log"
      id="talk-comments-stream-log"
      role="log"
    />
  </div>
</section>
`;

exports[`renders message box when logged in 1`] = `
<section
  aria-labelledby="tab-COMMENTS"
  className="App-tabContent"
  data-testid="current-tab-pane"
  id="tabPane-COMMENTS"
  role="tabpanel"
>
  <div
    className="HorizontalGutter-root Stream-root HorizontalGutter-double"
  >
    <div
      className="Flex-root"
    >
      <div
        className="Flex-flex Flex-halfItemGutter Flex-wrap"
      >
        <div
          className="Typography-root Typography-bodyCopy Typography-colorTextPrimary"
        >
          Signed in as 
          <span
            className="Typography-root Typography-bodyCopyBold Typography-colorTextPrimary"
          >
            Markus
          </span>
          .
        </div>
        <div
          className="Flex-root Typography-root Typography-bodyCopy Typography-colorTextPrimary Flex-flex"
        >
          <span>
            Not you?Â 
          </span>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign Out
          </button>
        </div>
      </div>
    </div>
    <div>
      <div
        className="MessageBox-root PostCommentForm-messageBox"
      >
        <span
          aria-hidden="true"
          className="Icon-root Icon-md MessageBoxIcon-root"
        >
          chat
        </span>
        <div
          className="Markdown-root MessageBoxContent-root"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong>What do you think</strong>?</p>
",
            }
          }
        />
      </div>
      <form
        autoComplete="off"
        id="comments-postCommentForm-form"
        onSubmit={[Function]}
      >
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-half"
          >
            <label
              className="AriaInfo-root"
              htmlFor="comments-postCommentForm-field"
            >
              Post a comment
            </label>
            <div>
              <div
                className=""
              >
                <div
                  aria-hidden="true"
                  className="RTE-placeholder RTE-placeholder"
                >
                  Post a comment
                </div>
                <div
                  aria-placeholder="Post a comment"
                  className="RTE-contentEditable RTE-content"
                  contentEditable={true}
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                  disabled={false}
                  id="comments-postCommentForm-field"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onCut={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyDown={[Function]}
                  onPaste={[Function]}
                  onSelect={[Function]}
                />
                <div
                  className="RTE-toolbar RTE-toolbarBottom Toolbar-toolbar"
                >
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Bold"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_bold
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Italic"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_italic
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Blockquote"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_quote
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            className="Flex-root Flex-flex Flex-alignFlexEnd Flex-directionColumn"
          >
            <button
              className="BaseButton-root Button-root Button-sizeRegular Button-colorPrimary Button-variantFilled Button-disabled"
              disabled={true}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      aria-live="polite"
      className="HorizontalGutter-root HorizontalGutter-full"
      data-testid="comments-stream-log"
      id="talk-comments-stream-log"
      role="log"
    />
  </div>
</section>
`;

exports[`renders message box when not logged in 1`] = `
<section
  aria-labelledby="tab-COMMENTS"
  className="App-tabContent"
  data-testid="current-tab-pane"
  id="tabPane-COMMENTS"
  role="tabpanel"
>
  <div
    className="HorizontalGutter-root Stream-root HorizontalGutter-double"
  >
    <div
      className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter"
    >
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Sign in
      </button>
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantOutlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
    <div>
      <div
        className="MessageBox-root PostCommentFormFake-messageBox"
      >
        <span
          aria-hidden="true"
          className="Icon-root Icon-md MessageBoxIcon-root"
        >
          chat
        </span>
        <div
          className="Markdown-root MessageBoxContent-root"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong>What do you think</strong>?</p>
",
            }
          }
        />
      </div>
      <div
        className="HorizontalGutter-root PostCommentFormFake-root HorizontalGutter-full"
      >
        <div
          aria-hidden="true"
        >
          <div>
            <div
              className=""
            >
              <div
                aria-hidden="true"
                className="RTE-placeholder RTE-placeholder "
              >
                Post a comment
              </div>
              <div
                aria-placeholder="Post a comment"
                className="RTE-contentEditable RTE-content  RTE-contentEditableDisabled"
                contentEditable={false}
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
                disabled={true}
                onBlur={[Function]}
                onChange={[Function]}
                onCut={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                onKeyDown={[Function]}
                onPaste={[Function]}
                onSelect={[Function]}
              />
              <div
                className="RTE-toolbar  RTE-toolbarDisabled RTE-toolbarBottom Toolbar-toolbar"
              >
                <button
                  className="Button-button"
                  disabled={false}
                  onClick={[Function]}
                  title="Bold"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="Icon-root Icon-md"
                  >
                    format_bold
                  </span>
                </button>
                <button
                  className="Button-button"
                  disabled={false}
                  onClick={[Function]}
                  title="Italic"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="Icon-root Icon-md"
                  >
                    format_italic
                  </span>
                </button>
                <button
                  className="Button-button"
                  disabled={false}
                  onClick={[Function]}
                  title="Blockquote"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    className="Icon-root Icon-md"
                  >
                    format_quote
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <button
          className="BaseButton-root Button-root Button-sizeRegular Button-colorPrimary Button-variantFilled Button-fullWidth Button-disabled"
          disabled={true}
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          type="submit"
        >
          Sign in and Join the Conversation
        </button>
      </div>
    </div>
    <div
      aria-live="polite"
      className="HorizontalGutter-root HorizontalGutter-full"
      data-testid="comments-stream-log"
      id="talk-comments-stream-log"
      role="log"
    />
  </div>
</section>
`;

exports[`renders message box when story isClosed 1`] = `
<section
  aria-labelledby="tab-COMMENTS"
  className="App-tabContent"
  data-testid="current-tab-pane"
  id="tabPane-COMMENTS"
  role="tabpanel"
>
  <div
    className="HorizontalGutter-root Stream-root HorizontalGutter-double"
  >
    <div
      className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter"
    >
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Sign in
      </button>
      <button
        className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantOutlined"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
    <div>
      <div
        className="MessageBox-root PostCommentFormClosed-messageBox"
      >
        <div
          className="Markdown-root MessageBoxContent-root"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p><strong>What do you think</strong>?</p>
",
            }
          }
        />
      </div>
      <div
        className="CallOut-root CallOut-colorRegular CallOut-fullWidth"
      >
        Story is closed
      </div>
    </div>
    <div
      aria-live="polite"
      className="HorizontalGutter-root HorizontalGutter-full"
      data-testid="comments-stream-log"
      id="talk-comments-stream-log"
      role="log"
    />
  </div>
</section>
`;
